### PR TITLE
fix how and when config is loaded

### PR DESCRIPTION
**Pull request by rjbs**
Without these two commits, there are two critical ways that things can fail:

* we try to use the config before we have set its location
* we try to use the sharedir config, which is never loaded properly

### CODE REVIEW

**Comment by msimerson** [comment]
I'd love to see some test coverage added the demonstrates and exercises the issue.

**Comment by rjbs** [comment]
Here's an amusing finding with trying to test the "fall back to sharedir."  

```perl
sub get_config {
    my $self = shift;
    my $file = shift || $self->{config_file} or croak;
    return Config::Tiny->read($file) if -r $file;  # fully qualified
```

…except just because a file can be read doesn't mean it's an absolute path.  In fact, it might just be a file in the cwd… and that's what it is when we're trying to load the default, because a file called `mail-dmarc.ini` exists in the repository.  If you remove that file, the code that's meant to load the default config in `t/03.Base.t` will fail, because it will enter the branch where it looks for the version in the sharedir, which returns a filename instead of config, which dies.

So, I've pushed a commit to remove that now-unneeded file.  Tests would have failed without my fix to get_config, and now pass.  No new tests seem needed on that front.

**Comment by rjbs** [comment]
This new test demonstrates problems with fully-random attribute initialization, if you remove the sorting up of config_file and run the test.  You may need to run it many times to get the "right" random order.

**Comment by rjbs** [comment]
That Travis failure was a good catch.  One test was still using the mail-dmarc.ini in the repo root instead of the sharedir, but my local testing didn't notice because I had Mail-DMARC installed.  Fixed with last commit, I think this branch is good.

File: t/00.Dmarc.t [hunk 1]
```diff
@@ -41,6 +41,26 @@ test_setter_values();
 test_spf();
 test_dkim();
 
+{ # config file loaded before any other attr initialization
```
Comment (msimerson) [inline t/00.Dmarc.t]: There's already a couple other package declarations in this file. Is there a reason not to place this one at the bottom of the file with the rest of them?

Also, is there a reason not to put this test into an aptly named sub as all the other tests in this file are?

**Comment by msimerson** [comment]
Besides my other comment, the only other thing I'd like to see before merging is squashing this branch down to 1 or 2 commits.

**Pull request closed without merging**